doc/architecture/platforms/snp.rst

```diff
@@ -0,0 +1,28 @@
+AMD SEV-SNP
+===================
+
+.. warning:: SEV-SNP support is currently experimental and in active development.
+
+How to use the AMD SEV-SNP platform
+-----------------------------------
+CCF must run on an AMD CPU which supports SEV-SNP.
+
+To use SNP, ``virtual`` must be specified in ``COMPILE_TARGETS``, at compile time. If any other targets are specified, you may also need to specify the ``TEST_ENCLAVE=virtual`` runtime argument.
+
+Attestation
+-----------
+SNP attestation provide several fields needed to establish trust.
+
+- Measurement
+
+Measurement is a digest of the UVM that is running, which CCF stores as a :doc:`code id <../../audit/builtin_maps>`. New nodes joining a network will provide their code id and the primary will perform an identity check against the table entries.
```

Comment:
We should be able to link directory to the `code id` table, rather than the `builtin_maps` document, using `:ref`.

Reply:
I tried to do that, I couldn't find a way for it to work with the style of headers on that page e.g.

```
``nodes.code_ids``
~~~~~~~~~~~~~~~~~~
```

Not sure if it's because of characters like backtick and period? I tried making my own anchor points such as
```
.. _nodes-code-ids:
```

But that didn't work either

Reply:
Ah! I remember spending some time trying to get this working. I think that **:ref:`audit/builtin_maps:``nodes.code_ids```** 
 (with all the backticks!) should work.

Reply:
Looks like we lose the ability to customise the text but I think that's okay
https://github.com/microsoft/CCF/pull/4330/commits/0e4c0d37dc1c8667829f8000e0f8b18a69886529